coupler/coupler_types.F90

```diff
@@ -1252,29 +1727,63 @@ subroutine CT_copy_data_2d(var_in, var, halo_size, bc_index, field_index,&
       j_off = var_in%jsc - var%jsc
     endif
 
-    do n = n1, n2
-      copy_bc = .true.
-      if (copy_bc .and. present(exclude_flux_type))&
-          & copy_bc = .not.(trim(var%bc(n)%flux_type) == trim(exclude_flux_type))
-      if (copy_bc .and. present(only_flux_type))&
-          & copy_bc = (trim(var%bc(n)%flux_type) == trim(only_flux_type))
-      if (copy_bc .and. present(pass_through_ice))&
-          & copy_bc = (pass_through_ice .eqv. var%bc(n)%pass_through_ice)
-      if (.not.copy_bc) cycle
+    if(var_in%set .and. var_in%num_bcs .gt. 0) then
+      if(associated(var_in%bc) .eqv. associated(var_in%bc_r4)) &
+        call mpp_error(FATAL, "CT_copy_data_2d"//err_msg_var_in_kind)
+    endif
 
-      do m = 1, var%bc(n)%num_fields
-        if (present(field_index)) then
-          if (m /= field_index) cycle
-        endif
-        if ( associated(var%bc(n)%field(m)%values) ) then
-          do j=var%jsc-halo,var%jec+halo
-            do i=var%isc-halo,var%iec+halo
-              var%bc(n)%field(m)%values(i,j) = var_in%bc(n)%field(m)%values(i+i_off,j+j_off)
+    ! num_bcs .lt. 1 -> loop doesn't run but shouldn't error out
+    if (associated(var_in%bc) .or. var_in%num_bcs .lt. 1) then
```

Comment:
Would it be clearer if var_in%num_bcs .lt. n1?

Reply:
Also wondering if var_in%num_bcs.lt.1 should be separated into its own if else block for clarity

Reply:
i think this is a little more clear. Some code within these is outside of loops so needs to run if either part of the conditional is true, otherwise will have to be repeated. could change the logic to use and .and. instead, but that would be more confusing